rules/TypeDeclaration/Collector/ParentClassCollector.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\Collector;
+
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Collectors\Collector;
+
+final class ParentClassCollector implements Collector
```

Comment:
should have a `@implements<Class_, class-string[]>` generic

Reply:
Done :+1:

Reply:
Hm, when I add it, it yield some contravariant bug I've never understood :D could you add it so it works well?

Reply:
it has to be

```diff
$ git diff
diff --git a/rules/TypeDeclaration/Collector/ParentClassCollector.php b/rules/TypeDeclaration/Collector/ParentClassCollector.php
index ad73832c5d..48125ca462 100644
--- a/rules/TypeDeclaration/Collector/ParentClassCollector.php
+++ b/rules/TypeDeclaration/Collector/ParentClassCollector.php
@@ -10,6 +10,9 @@ use PhpParser\Node\Stmt\Class_;
 use PHPStan\Analyser\Scope;
 use PHPStan\Collectors\Collector;

+/**
+ * @implements Collector<Class_, string[]>
+ */
 final class ParentClassCollector implements Collector
 {
     public function getNodeType(): string
@@ -19,7 +22,6 @@ final class ParentClassCollector implements Collector

     /**
      * @param Class_ $node
-     * @return mixed[]|null
      */
     public function processNode(Node $node, Scope $scope): ?array
     {

```

Reply:
Ah, got it :) thanks :+1: